test/e2e/smoke_test.go

```diff
@@ -0,0 +1,141 @@
+				GOOD_TEMPLATE, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create a secret to hold the template params
+			coreclient, err := corev1client.NewForConfig(c.config)
+			Expect(err).NotTo(HaveOccurred())
+			secret, err := coreclient.Secrets(c.namespace).Create(&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "parameters",
+				},
+				StringData: map[string]string{
+					"NAME":         GOOD_TEMPLATE,
+					"MEMORY_LIMIT": "1024Mi",
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Instantiate the template
+			ti, err := templateclient.TemplateInstances(c.namespace).Create(
+				&templatev1.TemplateInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "templateinstance",
+					},
+					Spec: templatev1.TemplateInstanceSpec{
+						Template: *template,
+						Secret: &corev1.LocalObjectReference{
+							Name: secret.Name,
+						},
+					},
+				})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for the deployments to finish
+			watcher, err := templateclient.TemplateInstances(c.namespace).Watch(
+				metav1.SingleObject(ti.ObjectMeta),
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			for event := range watcher.ResultChan() {
+				switch event.Type {
+				case watch.Modified:
+					ti = event.Object.(*templatev1.TemplateInstance)
+
+					for _, cond := range ti.Status.Conditions {
+						// If the TemplateInstance contains a status condition
+						// Ready == True, stop watching.
+						if cond.Type == templatev1.TemplateInstanceReady &&
+							cond.Status == corev1.ConditionTrue {
+							watcher.Stop()
+						}
+
+						// If the TemplateInstance contains a status condition
+						// InstantiateFailure == True, indicate failure.
+						if cond.Type ==
+							templatev1.TemplateInstanceInstantiateFailure &&
+							cond.Status == corev1.ConditionTrue {
+							panic("templateinstance instantiation failed")
+						}
+					}
+
+				default:
+					panic("unexpected event type " + event.Type)
+				}
+			}
+
+			r := regexp.MustCompile(REGEX)
+
+			// Pull the route ingress from the namespace
+			routeclient, err := routev1client.NewForConfig(c.config)
+			Expect(err).NotTo(HaveOccurred())
+			route, err := routeclient.Routes(c.namespace).Get(GOOD_TEMPLATE, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			host := route.Status.Ingress[0].Host
+			url := fmt.Sprintf("http://%s", host)
+
+			// Hit the ingress 3 times, each time the hit counter should increment
+			prevCounter := 0
+			currCounter := 0
+			for i := 1; i <= 3; i++ {
+				resp, err := http.Get(url)
+				Expect(err).NotTo(HaveOccurred())
+				defer resp.Body.Close()
+				contents, err := ioutil.ReadAll(resp.Body)
+				Expect(err).NotTo(HaveOccurred())
+				matches := r.FindStringSubmatch(string(contents))
+				currCounter, err = strconv.Atoi(matches[1])
+				Expect(err).NotTo(HaveOccurred())
+				Expect(currCounter).Should(BeNumerically(">", prevCounter))
+				prevCounter = currCounter
+				time.Sleep(1)
```

Comment:
This is not a second but a nanosecond. You probably want `time.Sleep(time.Second))`